From 19be7519e7fb0e02107653d866d2d6c53eac50c8 Mon Sep 17 00:00:00 2001
From: Hari Nagalla <hnagalla@ti.com>
Date: Mon, 29 Jul 2024 21:51:46 -0500
Subject: [PATCH] remoteproc: k3-dsp: assert reset in probe

Assert reset in probe. This ensures the DSP is in reset and does
not start running bogus code when remote proc driver invokes
prepare call to load firmware.

Signed-off-by: Hari Nagalla <hnagalla@ti.com>
---
 drivers/remoteproc/ti_k3_dsp_remoteproc.c | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/drivers/remoteproc/ti_k3_dsp_remoteproc.c b/drivers/remoteproc/ti_k3_dsp_remoteproc.c
index 071097a72cb1..635a534de68e 100644
--- a/drivers/remoteproc/ti_k3_dsp_remoteproc.c
+++ b/drivers/remoteproc/ti_k3_dsp_remoteproc.c
@@ -806,18 +806,16 @@ static int k3_dsp_rproc_probe(struct platform_device *pdev)
 	} else {
 		dev_info(dev, "configured DSP for remoteproc mode\n");
 		/*
-		 * ensure the DSP local reset is asserted to ensure the DSP
+		 * ensure the DSP reset is asserted to ensure the DSP
 		 * doesn't execute bogus code in .prepare() when the module
 		 * reset is released.
 		 */
-		if (data->uses_lreset) {
-			ret = reset_control_status(kproc->reset);
-			if (ret < 0) {
-				return dev_err_probe(dev, ret, "failed to get reset status\n");
-			} else if (ret == 0) {
-				dev_warn(dev, "local reset is deasserted for device\n");
-				k3_dsp_rproc_reset(kproc);
-			}
+		ret = reset_control_status(kproc->reset);
+		if (ret < 0) {
+			return dev_err_probe(dev, ret, "failed to get reset status\n");
+		} else if (ret == 0) {
+			dev_warn(dev, "local reset is deasserted for device\n");
+			k3_dsp_rproc_reset(kproc);
 		}
 	}
 
-- 
2.34.1

